File: 3. Largest Prime Factor/solution.ts
function largestPrimeFactor(n: number): number {
    let i: number = 2;
    while (i * i <= n) {
        while (n % i === 0) {
            n /= i;
        }
        i++;
    }
    return n;
}

const number: number = 600851475143;
const largestPrime: number = largestPrimeFactor(number);
console.log(largestPrime);

File: 3. Largest Prime Factor/solution.go
package main

import (
	"fmt"
)

func largestPrimeFactor(n int64) int64 {
	var i int64 = 2
	for i*i <= n {
		for n%i == 0 {
			n /= i
		}
		i++
	}
	return n
}

func main() {
	number := int64(600851475143)
	largestPrime := largestPrimeFactor(number)
	fmt.Println(largestPrime)
}
